segpy/binary_reel_header.py

```diff
@@ -1,5 +1,73 @@
 from segpy.header import FormatMeta, field
-from segpy.field_types import Int32, Int16
+from segpy.field_types import IntFieldMeta, IntEnumFieldMeta, Int32, Int16, UInt32, UInt16
+
+
+class DataSampleFormatField(metaclass=IntEnumFieldMeta,
```

Comment:
See my comments about `IntEnum` elsewhere. It feels like we're restating information which is already in `segpy.datatypes.DATA_SAMPLE_FORMAT_TO_SEG_Y_TYPE`.  Ideally I think `DATA_SAMPLE_FORMAT_TO_SEG_Y_TYPE` could be an `IntEnum` (or similar) and perhaps there's a good argument that is should be *this* enum. Maybe that would get too circular.

Reply:
I've actually made exactly this change in further work. It turns out to be necessary since for testing since we ultimately look this field up in `DATA_SAMPLE_FORMAT_TO_SEG_Y_TYPE`. I'll pull into here.